### PR TITLE
cmd: collect string utilities in one module, add missing tests

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

File: cmd/libsnap-confine-private/string-utils.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	size_t slen = strlen(str);
+
+	if (slen < xlen) {
+		return false;
+	}
+
+	return strncmp(str - xlen + slen, suffix, xlen) == 0;
+}
+
+int must_snprintf(char *str, size_t size, const char *format, ...)
```
Comment (niemeyer) [inline cmd/libsnap-confine-private/string-utils.c]: Should that be sc_?

Comment (zyga) [inline cmd/libsnap-confine-private/string-utils.c]: Yes but I didn't want to make the diff huge (it's a very popular function). I will do that separately.

File: cmd/libsnap-confine-private/string-utils-test.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	g_assert_false(sc_endswith("", NULL));
+	g_assert_false(sc_endswith(NULL, ""));
+	g_assert_false(sc_endswith(NULL, NULL));
+	g_assert_true(sc_endswith("", ""));
+	g_assert_true(sc_endswith("foobar", "bar"));
+	g_assert_true(sc_endswith("foobar", "ar"));
+	g_assert_true(sc_endswith("foobar", "r"));
+	g_assert_true(sc_endswith("foobar", ""));
+	g_assert_false(sc_endswith("foobar", "quux"));
+	g_assert_false(sc_endswith("", "bar"));
```
Comment (mvo5) [inline cmd/libsnap-confine-private/string-utils-test.c]: Maybe a test here in the spirit of line 41ff. Something like:
```
g_assert_false(sc_endswith("b", "bar"));
g_assert_false(sc_endswith("ba", "bar"));
g_assert_true(sc_endswith("bar", "bar"));
```
?

Comment (zyga) [inline cmd/libsnap-confine-private/string-utils-test.c]: Absolutely, thanks for spotting this!

**Pull request merged**